charts/surrealdb/templates/hpa.yaml

```
@@ -0,0 +1,19 @@
+{{- if .Values.horizontalPodAutoscaler}}
```

There could be various conventions here, but I believe the most popular is to have an explicit helm chart value to toggle the HPA:

```suggestion
{{- if .Values.autoscaling.enabled }}
```

---

It may not be necessary to add an autoscaling enabled flag. I can make the changes such that only when the horizontal pod autoscaler variables are defined, then the autoscaling feature is enabled, otherwise it uses replicaCount

---

Thanks for your response! Yeah, it may not be necessary. I just believe using the enabled flag pattern is better.

By using the existence of a non-empty value to enable the feature, you end up with multiple`.Values.foo | default ...` in the chart templates, like `minReplicas: {{ .minReplicas | default 1 }}` we can see in the diff. That's repetition because we write down the same default values in two other places, the values.yaml comments and README.

With the `.enabled` pattern, you can declare and document all the defaults in the single-place- chart's default values.yaml file- and we can generate the README.md from the values.yml using `helm-docs` in the future, leaving us no need to touch the defaults in the templates and manually update the docs.

---

Okay added `.Values.horizontalPodAutoscaler.enabled` pattern to be compatable with helm-docs